pkg/resources/management.cattle.io/v3/globalrolebinding/validator.go

```diff
@@ -102,9 +103,55 @@ func (a *admitter) Admit(request *admission.Request) (*admissionv1.AdmissionResp
 			Allowed: false,
 		}, nil
 	}
+	fieldPath := field.NewPath("globalrolebinding")
+	if request.Operation == admissionv1.Create {
+		// new bindings can't refer to a locked role template. The GR validator also checks this, but adding the check here
+		// allows the GR validator to permit updates to GRs using locked roleTemplates without removing the locked permission
+		err := a.validateGlobalRole(globalRole, fieldPath)
+		if err != nil {
+			if fieldError, ok := err.(*field.Error); ok {
+				return admission.ResponseBadRequest(fieldError.Error()), nil
+			}
+			return nil, err
+		}
+	}
 
-	response := &admissionv1.AdmissionResponse{}
-	auth.SetEscalationResponse(response, auth.ConfirmNoEscalation(request, globalRole.Rules, "", a.resolver))
+	clusterRules, err := a.grbResolver.GlobalRoleResolver.ClusterRulesFromRole(globalRole)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			reason := fmt.Sprintf("at least one roleTemplate was not found %s", err.Error())
+			return admission.ResponseBadRequest(reason), nil
+		}
+		return nil, fmt.Errorf("unable to get global rules from role %s: %w", globalRole.Name, err)
+	}
+	err = auth.ConfirmNoEscalation(request, clusterRules, "", a.grbResolver)
+	if err != nil {
+		return admission.ResponseFailedEscalation(err.Error()), nil
+	}
 
-	return response, nil
+	rules := a.grbResolver.GlobalRoleResolver.GlobalRulesFromRole(globalRole)
+	err = auth.ConfirmNoEscalation(request, rules, "", a.resolver)
+	if err != nil {
+		return admission.ResponseFailedEscalation(err.Error()), nil
+	}
+	return admission.ResponseAllowed(), nil
+}
+
+// validateGlobalRole validates that the attached global role isn't trying to use a locked RoleTemplate.
+func (a *admitter) validateGlobalRole(globalRole *v3.GlobalRole, fieldPath *field.Path) error {
+	roleTemplates, err := a.grbResolver.GlobalRoleResolver.GetRoleTemplatesForGlobalRole(globalRole)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			reason := fmt.Sprintf("unable to find all roleTemplates %s", err.Error())
+			return field.Invalid(fieldPath, "", reason)
+		}
+		return fmt.Errorf("unable to get role templates for global role %s: %w", globalRole.Name, err)
+	}
+	for _, roleTemplate := range roleTemplates {
+		if roleTemplate.Locked {
+			reason := fmt.Sprintf("global role inherits roleTemplate %s which is locked", roleTemplate.Name)
+			return field.Invalid(fieldPath.Child("globalRoleName"), globalRole.Name, reason)
+		}
+	}
```

Comment:
As multiple role templates could be locked, what if the `for` aggregates all the locked RT names into a variable, and after the `for` we return the aggregated list of locked RT names? The reasoning being that the short-circuit does not save much time/space as the check is inexpensive and all the data is already in memory.

Reply:
Yep, that makes sense and I think produces a far better UX.

Reply:
Done.